cmd/hope/deploy.go

```diff
@@ -100,14 +100,19 @@ var deployCmd = &cobra.Command{
 				return err
 			}
 
+			parameters, err := utils.RenderParameters(resource.Parameters, resource.FileParameters)
+			if err != nil {
+				return err
+			}
+
 			switch resourceType {
 			case hope.ResourceTypeFile:
-				if len(resource.Parameters) != 0 {
+				if len(parameters) != 0 {
```

Comment:
The depth of blocks here, and similarities with the delete cli makes me feel like this is getting dangerously close to needing to be pulled out into a separate batch of types + interface. The `Resource` interface would ask its types to implement `Deploy`, `Delete`, and... others?

Reply:
In this case, the resource will be able to own its own parameter management logic too, but then resources will also need to learn how to deal with files and whatnot, which may be a bit excessive.

Reply:
Created #63 to deal with this later.